alert.go

```diff
@@ -910,6 +916,31 @@ func (a *AlertNode) handlePost(post *pipeline.PostHandler, ad *AlertData) {
 	return
 }
 
+func (a *AlertNode) handleTcp(tcp *pipeline.TcpHandler, ad *AlertData) {
+	bodyBuffer := a.bufPool.Get().(*bytes.Buffer)
+	defer func() {
+		bodyBuffer.Reset()
+		a.bufPool.Put(bodyBuffer)
+	}()
+
+	err := json.NewEncoder(bodyBuffer).Encode(ad)
+	if err != nil {
+		a.logger.Println("E! failed to marshal alert data json", err)
+		return
+	}
+
+	conn, err := net.Dial("tcp", tcp.Address)
+	if err != nil {
+		a.logger.Println("E! failed to connect", err)
+		return
+	}
+
+	fmt.Fprintf(conn, bodyBuffer+"\n")
+
+	defer conn.Close()
```

Comment:
Actually I was incorrect, and the `defer` has now been moved to be before any data is written over the connection.